ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -635,169 +568,42 @@ sub Configure
-        if ( $cmd_exit ) {
-            $self->error("Error updating packages:\n$cmd_err");
-            return 0;
-        }
-        $installed = $self->get_installed_rpms();
-        return 1 if !defined($installed);
-        my $updated_to = $installed - $pre;
-        if ( !$updated_to->is_empty ) {
-            $self->info("updated ".$updated_to->size." package(s) to: ", $updated_to);
-        }
-    } else {
-        $self->info("no packages to upgrade/downgrade");
+    # Print summary what is supposed to be done.
+    $self->info("Transaction summary --------------------");
+    if ( defined($whitelisted) && scalar @$whitelisted > 0 ) {
+        $self->info( "whitelist ", scalar @$whitelisted, " package(s): ", join ( " ", sort @$whitelisted ) );
     }
-
-    # Install only additional packages to what is already installed
-    my $will_install = $to_install - $installed;
-    if ( !$will_install->is_empty ) {
-        my $pre = $installed;
-        ( $cmd_exit, $cmd_out, $cmd_err ) = $self->execute_yum_with_recovery( [ "yum install -y " . YUM_PLUGIN_OPTS . " " . join( " ", sort @$will_install) ], "installing ".$will_install->size." package(s)" );
-        if ( index($cmd_out, 'Complete!') == -1 ) {
-            $self->error("Error installing packages:\n$cmd_err");
-            return 0;
-        }
-        $installed = $self->get_installed_rpms();
-        return 1 if !defined($installed);
-        my $installed_new = $installed - $pre;
-        if ( !$installed_new->is_empty ) {
-            $self->info("installed " . $installed_new->size . " package(s): ", $installed_new);
-        }
-    } else {
-        $self->info("nothing to install");
+    if (@$excludes) {
+        $self->info( "exclude   ", scalar @$excludes, " package(s): ", join ( " ", sort @$excludes ) );
     }
+    $self->info( "install   " , $will_install->size, " package(s): ", join ( " ", sort @$will_install ) );
+    $self->info( "remove    ", $will_remove->size, " package(s): ", join ( " ", sort @$will_remove ) );
+    $self->info("----------------------------------------");
 
-    # Create list of packages to be removed
-    my $will_remove = $installed - $to_install;
-    my $whitelisted = Set::Scalar->new();
-    $whitelist = $t->{whitelist};
-    for my $rpm ( $will_remove->elements ) {    # do not remove imported GPG keys
-        if ( substr( $rpm, 0, 13 ) eq '0:gpg-pubkey-' ) {
-            $will_remove->delete($rpm);
-        }
-        # Do not remove whitelisted packages.
-        if (defined($whitelist)) {
-            for my $white_pkg (@$whitelist) {
-                my $rpm_noepoch = $rpm;
-                $rpm_noepoch =~ s/^.*://;
-                if ( index($rpm_noepoch, $white_pkg) == 0 || match_glob($white_pkg, $rpm_noepoch) ) {
-                    $will_remove->delete($rpm);
-                    $whitelisted->insert($rpm);
-                }
-            }
-        }
+    # End here in case of --noaction.
+    if ($NoAction) {
+        return 1;
```

Comment:
add a info message of some sort

Reply:
The message is already there:

[INFO] Dry run, no changes will be performed (--noaction flag set)

Reply:
and:

[INFO] running component: spma
---------------------------------------------------------
[INFO] would mark state of component as needing to run
[INFO] note: running component spma in noaction mode

in the component directly. Assuming we don't need to add this message for the third time in the log.